### PR TITLE
chore(TS): fix type of modifed event that could cause unexpected behaviour in dev code

### LINKED ISSUE #9498: `endCurrentTransform` severe bug
https://github.com/fabricjs/fabric.js/blob/44a75a3a7d9acd7d09d3ce4f5b0a92ec49d0680b/src/canvas/SelectableCanvas.ts#L1182-L1183

This is a bug because e is optional but `endCurrentTransform` does not accept an optional param causing a exception

### PR DESCRIPTION
**Pull request by asturur**
## Description

As #9500 but without refactors of code

close #9498

### CODE REVIEW

File: src/canvas/SelectableCanvas.ts [hunk 1]
```diff
@@ -1188,6 +1187,48 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
...
+    }
+
+    target.setCoords();
+
+    if (transform.actionPerformed) {
+      this.fire('object:modified', options);
+      target.fire('modified', options);
+    }
+  }
+
```
Comment (asturur) [inline src/canvas/SelectableCanvas.ts:1231]: moved so typescript can check

**Comment by ShaMan123** [comment]
> #### Review or Edit in CodeSandbox
> Open the branch in [Web Editor](https://codesandbox.io/p/github/fabricjs/fabric.js/fix-9498-no-refactor?mode=review&utm_source=gh_app) • [VS Code](https://codesandbox.io/p/vscode?owner=fabricjs&repo=fabric.js&branch=fix-9498-no-refactor&utm_source=gh_app) • [Insiders](https://codesandbox.io/p/vscode?owner=fabricjs&repo=fabric.js&branch=fix-9498-no-refactor&insiders=true&utm_source=gh_app)
> 
> Open Preview
> * [Run Next Template](https://codesandbox.io/p/devtool/preview/fabricjs/fabric.js/fix-9498-no-refactor?task=next&port=3000&redirect=true&utm_source=gh_app)
> * [Run Node Template](https://codesandbox.io/p/devtool/preview/fabricjs/fabric.js/fix-9498-no-refactor?task=node&port=8080&redirect=true&utm_source=gh_app)
> * [Run Vanilla Template](https://codesandbox.io/p/devtool/preview/fabricjs/fabric.js/fix-9498-no-refactor?task=vanilla&port=1234&redirect=true&utm_source=gh_app)

@asturur did you see this? WOW! You made it work?

**Comment by asturur** [comment]
i got an email at some point 2-3 days ago asking me to enable something and i enabled it.

**Comment by ShaMan123** [comment]
I re triggered the request

**Comment by ShaMan123** [comment]
something does not work though, the links don't do anything after being opened

**Pull request merged**